src/NHibernate/Impl/ExpressionQueryImpl.cs

```diff
@@ -160,6 +160,7 @@ public ExpandedQueryExpression(IQueryExpression queryExpression, IASTNode tree,
 			Key = key;
 			Type = queryExpression.Type;
 			ParameterDescriptors = queryExpression.ParameterDescriptors;
+			CanCachePlan = (queryExpression as ICacheableQueryExpression)?.CanCachePlan ?? true;
```

Comment:
Defaults to `true` for consistency with existing checks in `GetHQLQueryPlan`/`GetFilterQueryPlan`

Reply:
Resolving `CanCachePan` in constructor could lead to some future bug, by example if someday the query translation, which mutates `CanCachePlan`, is moved after constructing this. (Granted, it is very unlikely since the constructor need the ast tree.)

But from a robustness standpoint, it would be better to have this evaluated in the property call.

Reply:
Ok. Moved to property getter